Add tests to custom makemessages command

chosak opened the pull request:
This change adds a Python test for the v1 app's custom `makemessages` Django management command, which overrides the default one documented at

https://docs.djangoproject.com/en/1.11/ref/django-admin/#django-admin-makemessages

Our custom makemessages command adds support for block tags like this:

```
{% trans %}
Multi-line
statement
here
{% endtrans %}
```

which is not supported in Django templates.

This change also removes support in this command for Django 1.8 now that this project is solely targeted at Django 1.11.

## Testing

To test this locally, you can verify that the unit test passes, and then remove our custom makemessages command (including the .pyc file if on Python 2). The built-in command will fail as it does not detect the multi-line Jinja2 tag.

Additionally, you can run through the makemessages workflow documented [here](https://cfpb.github.io/cfgov-refresh/translation/) to confirm that the command still works.

## Checklist

- [X] PR has an informative and human-readable title
- [X] Changes are limited to a single goal (no scope creep)
- [X] Code can be automatically merged (no conflicts)
- [X] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [X] Passes all existing automated tests
- [X] Any _change_ in functionality is tested
- [X] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Project documentation has been updated
- [X] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

schbetsy commented:
When I run `tox -r -e fast` I get an error on the `makemessages` test:

```
CommandError: Can't find msguniq. Make sure you have GNU gettext tools 0.15 or newer installed.
```

But I seem to have an up-to-date gettext installed (unless these are referring to different tools?):

```
> brew upgrade gettext
Error: gettext 0.19.8.1 already installed
```

chosak commented:
@schbetsy quite honestly I can't remember, but I do have that `brew link gettext --force` way back in my bash history. This is my local version:

```sh
$ gettext --version
gettext (GNU gettext-runtime) 0.19.8.1
```

schbetsy commented:
Okay, after running `brew link gettext --force`, this now works for me. How would you feel about adding that to the documentation somewhere before we merge? 

I get the feeling we'll get a lot of questions about it after we merge, since people won't be able to run the unit tests locally without doing that command. Should we do anything else to communicate the change?

schbetsy commented:
Yep, I like that plan! And I think the documentation you proposed looks good.

chosak commented:
@schbetsy I've updated the documentation and added logic to skip the translation test without gettext.

---

The documentation looks like this when rendered:

![image](https://user-images.githubusercontent.com/654645/46226511-81e34c80-c32a-11e8-9dcb-7b2281ee0b46.png)

![image](https://user-images.githubusercontent.com/654645/46226570-b525db80-c32a-11e8-85b6-7dce9d8b9e0d.png)

---

The tests will be skipped if you don't have gettext:

```sh
$ brew unlink gettext
$ tox -e fast v1.tests.management.commands.test_makemessages
s
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK (skipped=1)
```

and pass if you do:

```sh
$ brew link --force gettext
$ tox -e fast v1.tests.management.commands.test_makemessages
.
----------------------------------------------------------------------
Ran 1 test in 0.092s

OK
```
